packages/agent-utils/src/__tests__/localVectorDB.spec.ts

```diff
@@ -25,6 +24,71 @@ const generateRandomString = (length: number): string => {
 }
 
 describe('Local Vector DB', () => {
+  it('should list created collections', async () => {
+    const workspace = new InMemoryWorkspace()
+    const env = new Env(process.env);
+    const consoleLogger = new ConsoleLogger();
+    const embeddingApi = new OpenAIEmbeddingAPI(
+      env.OPENAI_API_KEY,
+      consoleLogger,
+      cl100k_base
+    )
+    const db = new LocalVectorDB(workspace, "testdb", embeddingApi)
+    db.addCollection("testA")
+    db.addCollection("testB")
+    
+    const collections = db.listCollections()
+    const names = collections.map(collection => collection.name)
+    const uris = collections.map(collection => collection.uri)
+
+    expect(names.length).toEqual(2)
+    expect(names).toContain("testA")
+    expect(names).toContain("testB")
+
+    expect(uris.length).toEqual(2)
+    expect(uris).toContain("testdb/testA")
+    expect(uris).toContain("testdb/testB")
+  })
+
+  it('should properly separate collection data', async () => {
+    const workspace = new InMemoryWorkspace()
+    const env = new Env(process.env);
+    const consoleLogger = new ConsoleLogger();
+    const embeddingApi = new OpenAIEmbeddingAPI(
+      env.OPENAI_API_KEY,
+      consoleLogger,
+      cl100k_base
+    )
+    const db = new LocalVectorDB(workspace,  "testdb", embeddingApi)
+    const collectionA = db.addCollection("testA")
+    const collectionB = db.addCollection("testB")
+
+    await collectionA.add([
+      "Hello world",
+      "Hello universe",
+    ])
+
+    await collectionB.add([
+      "Goodbye world",
+      "Goodbye universe",
+    ])
+
+    const resultsA = await collectionA.search("Hey", 4)
+    const textsA = resultsA.map(result => result.text())
+
+    const resultsB = await collectionB.search("Hey", 4)
```

Comment:
Don't you want to search for something exclusively that's contained within B and not contained with A?

I feel like a better example could be:
```
collectionA = ["dog", "cat", "bird"]
collectionB = ["right", "left", "south"]

collectionA.search("animal").expectLength(3)
collectionB.search("animal").expectLength(0)

collectionA.search("direction").expectLength(0)
collectionB.search("direction").expectLength(3)
```

Reply:
Yeah, I get what you mean but in order for that behavior to happen, we'd need to add a "similarity threshold". If no scores are above the threshold, then don't return any. The current behavior is "sort vectors by similarity and return top k".

Doing that isn't trivial and not all VectorDBs have something like that (https://community.pinecone.io/t/score-threshold-to-filter-out-irrelevant-results/966)